torchquad/integration/gaussian.py

```diff
@@ -0,0 +1,209 @@
+            x=do("repeat",x.cpu().detach(),self._dim,like="numpy").reshape(self._dim,x.shape[0]).to(x.device)
+        return x
+    
+    def integrate(self, fn, dim, args=None, N=8, integration_domain=None):
+        """Integrates the passed function on the passed domain using fixed-point Gaussian quadrature.
+
+        Args:
+            fn (func): The function to integrate over.
+            dim (int): Dimensionality of the function to integrate.
+            args (iterable object, optional): Additional arguments ``t0, ..., tn``, required by `fn`.
+            N (int, optional): Degree to use for computing sample points and weights. Defaults to 8.
+            integration_domain (list, optional): Integration domain, e.g. [[-1,1],[0,1]]. Defaults to [-1,1]^dim.
+
+        Returns:
+            float: integral value
+
+        """
+        if not integration_domain or self.name in ["Gauss-Laguerre","Gauss-Hermite"]:
+            integration_domain=self.default_integration_domain*dim
+            if self.name in ["Gauss-Laguerre","Gauss-Hermite"]:
+                logger.info(f"{self.name} integration only allowed over the interval {self.default_integration_domain[0]}!")
+                
+        self._integration_domain = _setup_integration_domain(dim, integration_domain)
+        self._check_inputs(dim=dim, N=N, integration_domain=self._integration_domain)
+            
+        logger.debug(f"Using {self.name} for integrating a fn with {N} points over {self._integration_domain}")
+
+        self._dim = dim
+        self._fn = fn
+        const=self._get_constant_multiplier()
+        
+        root_args=(N,)+self.root_args
+        
+        xi, wi = self._points_and_weights(self.root_fn,root_args,wrapper_func=self.wrapper_func)
+
+        integral= anp.sum(const*anp.sum(self._eval(xi,args=args,weights=wi),axis=1)) #what if there is a sum in the function? then wi*self._eval() will have dimension mismatch
+        logger.info(f"Computed integral was {integral}.")
+
+        return integral
+
+
+class GaussLegendre(Gaussian):
+    """Gauss Legendre quadrature rule in torch. See https://en.wikipedia.org/wiki/Gaussian_quadrature#Gauss%E2%80%93Legendre_quadrature.
+    
+    Examples
+    --------
+    >>> gl=torchquad.GaussLegendre()
+    >>> integral = gl.integrate(lambda x:np.sin(x), dim=1, N=101, integration_domain=[[0,5]]) #integral from 0 to 5 of np.sin(x)
+    |TQ-INFO| Computed integral was 0.7163378000259399 #analytic result = 1-np.cos(5)"""
+
+    def __init__(self):
+        super().__init__()
+        
+    def wrapper_func(self,xi,wi):  #scale from [-1,1] to [a,b]
+        a,b=self._integration_domain.T
+        x0 = 0.5 * (1.0 - xi) #self.points)
+        x1 = 0.5 * (1.0 + xi)#self.points)
+        xi = anp.outer(a, x0) + anp.outer(b, x1)
+        return xi,wi
+        
+    def _get_constant_multiplier(self): #do this somewhere else? with points and weights maybe?
```

Comment:
could you add a docstring to this method? also the name seems very cryptic, if another less generic name would be possible would be great too